crates/ruff_python_formatter/src/statement/stmt_assign.rs

```diff
@@ -1,12 +1,58 @@
-use crate::{verbatim_text, FormatNodeRule, PyFormatter};
-use ruff_formatter::{write, Buffer, FormatResult};
+use crate::context::PyFormatContext;
+use crate::{AsFormat, FormatNodeRule, PyFormatter};
+use ruff_formatter::formatter::Formatter;
+use ruff_formatter::prelude::text;
+use ruff_formatter::{write, Buffer, Format, FormatResult};
+use ruff_python_ast::prelude::Expr;
 use rustpython_parser::ast::StmtAssign;
 
 #[derive(Default)]
 pub struct FormatStmtAssign;
 
 impl FormatNodeRule<StmtAssign> for FormatStmtAssign {
     fn fmt_fields(&self, item: &StmtAssign, f: &mut PyFormatter) -> FormatResult<()> {
-        write!(f, [verbatim_text(item.range)])
+        let StmtAssign {
+            range: _,
+            targets,
+            value,
+            type_comment: _,
+        } = item;
+        write!(
+            f,
+            [LhsAssignList::new(targets), text(" = "), value.format()]
```

Comment:
Should this use value.format().with_options(Parenthesize::IfBreaks) to only add parentheses if necessary? Or what's black's behavior?

Reply:
I couldn't find a symbol like `Parenthesize::IfBreaks`, i now added a group and if_group_breaks which doesn't do the same as black but still at least keeps the line length constraint

Reply:
Yeah, this is in the binary expression formatting

Reply:
i realized i can change both PRs significantly with the new utils